core/services/ocr2/plugins/mercury/plugin.go

```diff
@@ -22,11 +28,36 @@ import (
 	mercuryv1 "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/mercury/v1"
 	mercuryv2 "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/mercury/v2"
 	mercuryv3 "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/mercury/v3"
+	"github.com/smartcontractkit/chainlink/v2/plugins"
 )
 
 type Config interface {
 	MaxSuccessfulRuns() uint64
 	ResultWriteQueueDepth() uint64
+	plugins.RegistrarConfig
+}
+
+// concrete implementation of MercuryConfig
+type mercuryConfig struct {
+	jobPipelineMaxSuccessfulRuns     uint64
+	jobPipelineResultWriteQueueDepth uint64
+	plugins.RegistrarConfig
+}
+
+func NewMercuryConfig(jobPipelineMaxSuccessfulRuns uint64, jobPipelineResultWriteQueueDepth uint64, pluginProcessCfg plugins.RegistrarConfig) Config {
+	return &mercuryConfig{
+		jobPipelineMaxSuccessfulRuns:     jobPipelineMaxSuccessfulRuns,
+		jobPipelineResultWriteQueueDepth: jobPipelineResultWriteQueueDepth,
+		RegistrarConfig:                  pluginProcessCfg,
+	}
+}
+
+func (m *mercuryConfig) MaxSuccessfulRuns() uint64 {
+	return m.jobPipelineMaxSuccessfulRuns
+}
+
+func (m *mercuryConfig) ResultWriteQueueDepth() uint64 {
+	return m.jobPipelineResultWriteQueueDepth
 }
 
 func NewServices(
```

Comment:
I would highly suggest breaking each case into a separate function to reduce the cognitive load of the `NewServices` function in total and to reduce the possibility of variable shadowing, which you're using naming to avoid (err, err2). My opinion is this produces cleaner code.

Reply:
good call out. i had considered this but took the lazier approach. agree that it reads better now